app/src/main/java/com/example/plantsapp/presentation/ui/tasks/TasksFragment.kt

```diff
@@ -47,10 +48,21 @@ class TasksFragment : Fragment(R.layout.fragment_tasks) {
         super.onViewCreated(view, savedInstanceState)
 
         with(tasksViewModel) {
-            loadingDialog.connectWith(isLoading, viewLifecycleOwner)
-
-            plantsWithTasks.observe(viewLifecycleOwner) {
-                plantsWithTasksAdapter.submitList(it)
+            tasksUiState.observe(viewLifecycleOwner) { state ->
+                when (state) {
+                    is TasksViewModel.TasksUiState.Loading -> {
+                        loadingDialog.show()
+                        loadingDialog.hideOnLifecycle(viewLifecycleOwner)
```

Comment:
1. It will be better to connect this not in some special state(like TasksViewModel.TasksUiState.Loading ) but little bit earlier